### PR TITLE
Cleanup in the implicit free surface tests + add `matrix_implicit_free_surface_solver_tests`

### LINKED ISSUE #2184: Should the `free_surface.implicit_step_solver.right_hand_side` be a vector in `HydrostaticFreeSurfaceModel` with a matrix solver?
Is this how it should be?

Looks like the right-hand-side of the free surface solver when `solver_method=:HeptadiagonalIterativeSolver` is used is not a Field but rather an array...

```Julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.5 (2021-12-19)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

julia> using Oceananigans
[ Info: Oceananigans will use 12 threads

julia> using Oceananigans.Units

julia> arch = CPU()

julia> grid = RectilinearGrid(arch, size = (128, 1, 5),
                                    x = (0, 1000kilometers), y = (0, 1), z = (-400, 0),
                                    topology = (Bounded, Periodic, Bounded))
128×1×5 RectilinearGrid{Float64, Bounded, Periodic, Bounded} on CPU with 1×1×1 halo
├── Bounded  x ∈ [0.0, 1.0e6]  regularly spaced with Δx=7812.5
├── Periodic y ∈ [0.0, 1.0)    regularly spaced with Δy=1.0
└── Bounded  z ∈ [-400.0, 0.0] regularly spaced with Δz=80.0

julia> model_pcg = HydrostaticFreeSurfaceModel(grid = grid,
                                               momentum_advection = nothing,
                                               free_surface = ImplicitFreeSurface(solver_method=:PreconditionedConjugateGradient,
                                                                                  tolerance = 1e-15))
HydrostaticFreeSurfaceModel{CPU, Float64}(time = 0 seconds, iteration = 0)
├── grid: 128×1×5 RectilinearGrid{Float64, Bounded, Periodic, Bounded} on CPU with 1×1×1 halo
├── tracers: (:T, :S)
├── closure: Nothing
├── buoyancy: Buoyancy{SeawaterBuoyancy{Float64, LinearEquationOfState{Float64}, Nothing, Nothing}, Oceananigans.Grids.ZDirection}
└── coriolis: Nothing

julia> model_mat = HydrostaticFreeSurfaceModel(grid = grid,
                                               momentum_advection = nothing,
                                               free_surface = ImplicitFreeSurface(solver_method=:HeptadiagonalIterativeSolver,
                                                                                  tolerance = 1e-15))
HydrostaticFreeSurfaceModel{CPU, Float64}(time = 0 seconds, iteration = 0)
├── grid: 128×1×5 RectilinearGrid{Float64, Bounded, Periodic, Bounded} on CPU with 1×1×1 halo
├── tracers: (:T, :S)
├── closure: Nothing
├── buoyancy: Buoyancy{SeawaterBuoyancy{Float64, LinearEquationOfState{Float64}, Nothing, Nothing}, Oceananigans.Grids.ZDirection}
└── coriolis: Nothing

julia> typeof(model_pcg.free_surface.implicit_step_solver.right_hand_side)
Field{Center, Center, Nothing, Nothing, RectilinearGrid{Float64, Bounded, Periodic, Bounded, Float64, Float64, Float64, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}, CPU}, Float64, OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, FieldBoundaryConditions{BoundaryCondition{Oceananigans.BoundaryConditions.Flux, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Flux, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Periodic, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Periodic, Nothing}, Nothing, Nothing, BoundaryCondition{Oceananigans.BoundaryConditions.Flux, Nothing}}, Nothing}

julia> typeof(model_mat.free_surface.implicit_step_solver.right_hand_side)
Vector{Float64} (alias for Array{Float64, 1})
```

If this is OK then we need to drop `interior()` from the `right_hand_side` at:
https://github.com/CliMA/Oceananigans.jl/blob/4f730161c8ea189779eed8c3b56961065f7699d3/test/test_implicit_free_surface_solver.jl#L111

cc: @simone-silvestri

### PR DESCRIPTION
**Pull request by navidcy**
Some cleaning up in the implicit free surface solvers tests.

Resolves #2184.

### CODE REVIEW

**Comment by navidcy** [comment]
@simone-silvestri, seems like

https://github.com/CliMA/Oceananigans.jl/blob/50c4b430bb1e6ed4aacaaf588943ada87a56aa7f/test/test_implicit_free_surface_solver.jl#L76-L177

is not used. Should we nuke it?

Also it seems to include deprecated syntax (i.e., `HydrostaticFreeSurfaceModel` no longer has an `architecture` keyword arg).

**Comment by simone-silvestri** [comment]
Ah, I didn't realize it was not used... If it is not too heavy I would keep it and maybe remove the `run_pcg_implicit_free_surface_solver_tests`. I am also ok with nuking it as implicit free surface solvers are compared in `test_hydrostatic_free_surface_immersed_boundaries_congrad_solve.jl` (which probably can be renamed to `test_hydrostatic_free_surface_immersed_boundaries_implicit_solve.jl`

**Comment by navidcy** [comment]
> I am also ok with nuking it as implicit free surface solvers are compared in `test_hydrostatic_free_surface_immersed_boundaries_congrad_solve.jl` (which probably can be renamed to `test_hydrostatic_free_surface_immersed_boundaries_implicit_solve.jl`)

but with immersed boundaries the fft solver is not compared, isn't it?

**Comment by navidcy** [comment]
@simone-silvestri, I added `run_matrix_implicit_free_surface_solver_tests` but kept `run_pcg_implicit_free_surface_solver_tests`. Is this an overkill?

File: test/test_implicit_free_surface_solver.jl [hunk 1]
```diff
@@ -187,6 +186,7 @@ end
             # This behavior is not observed on tartarus, where this test _would_ pass.
             # Suffice to say that the FFT solver appears to be accurate (as of this writing), and tests pass
             # on the CPU.
+            @info "  skipping comparisson between pcg and fft implicit free surface solver comparison"
```
Comment (glwagner) [inline test/test_implicit_free_surface_solver.jl:189]:
```suggestion
            @info "  Skipping comparison between pcg and fft implicit free surface solver"
```

**Review by glwagner — APPROVED** [review]

**Comment by simone-silvestri** [comment]
I think it's ok, the hydrostatic model tests are not the bottleneck

**Review by simone-silvestri — APPROVED** [review]

**Pull request merged**